### PR TITLE
Add a function to compute the spatial intersection of a segment with a VoxelData object

### PR DESCRIPTION
**Pull request by adrien-berchet**
Following #5

### CODE REVIEW

**Comment by adrien-berchet** [comment]
@mgeplf I think it's good now

File: voxcell/voxel_data.py [hunk 1]
```diff
@@ -205,7 +205,11 @@ def positions_to_indices(self, positions, strict=True, keep_fraction=False):
         if not keep_fraction:
             result = np.floor(result).astype(int)
         result[result < 0] = VoxelData.OUT_OF_BOUNDS
-        result[result >= self.shape] = VoxelData.OUT_OF_BOUNDS
+        result[(result >= self.shape) & (positions >= self.bbox[1])] = VoxelData.OUT_OF_BOUNDS
```
Comment (mgeplf) [inline voxcell/voxel_data.py:208]: Really nice find!

I think I need to break this out into a separate PR, and go over the other code that does similar calcs - do you mind? It might take a couple days

File: tox.ini [hunk 2]
```diff
@@ -13,7 +13,7 @@ envlist =
 [testenv]
 extras = all
 deps = {[base]testdeps}
-commands = pytest tests {posargs}
```
[the remark this answers is not in the document]
Comment (adrien-berchet) [reply tox.ini:16]: This is to be able to run only one test with tox with a command like `tox -e py38 -- tests/test_math_utils.py::the_name_of_the_test`. But I forgot to not commit it, we can discard it if you prefer.

A cleaner war to retrict the paths in which `pytest` searches for tests is to add this in the `pyproject.toml` file:
```ini
[tool.pytest.ini_options]
testpaths = [
    "tests",
]
```
So when no test path is given to pytest, it will search for all tests in the `tests` directory, while if test paths are given it will only consider there tests.

File: voxcell/voxel_data.py (hunk 1 above)
Comment (adrien-berchet) [reply voxcell/voxel_data.py:208]: Sure, no problem, it's not so urgent. And if you need help for the new PR I can take some more time for this.

File: tox.ini (hunk 2 above)
[the remark this answers is not in the document]
Comment (mgeplf) [reply tox.ini:16]: cool, I'll apply that, tnx.

File: voxcell/voxel_data.py (hunk 1 above)
Comment (mgeplf) [reply voxcell/voxel_data.py:210]: ok, I don't have time to find a more elegant way of doing this; we'll just keep this for now

**Review by mgeplf — APPROVED** [review]

File: tox.ini (hunk 2 above)
[the remark this answers is not in the document]
Comment (adrien-berchet) [reply tox.ini:16]: you're welcome :)

**Comment by adrien-berchet** [comment]
Thanks @mgeplf !
